armi/reactor/converters/tests/test_axialExpansionChanger.py

```diff
@@ -722,103 +688,148 @@ class TestDetermineTargetComponent(AxialExpansionTestBase, unittest.TestCase):
         fuel = Circle("fuel", "FakeMat", **fuelDims)
         poison = Circle("poison", "FakeMat", **poisonDims)
         b.add(fuel)
         b.add(poison)
+        b.add(self.coolant)
         # call method, and check that target component is correct
-        self.obj.expansionData.determineTargetComponent(b)
+        self.expData.determineTargetComponent(b)
         self.assertTrue(
-            self.obj.expansionData.isTargetComponent(fuel),
-            msg="determineTargetComponent failed to recognize intended component: {}".format(
-                fuel
-            ),
+            self.expData.isTargetComponent(fuel),
+            msg=f"determineTargetComponent failed to recognize intended component: {fuel}",
         )
 
-    def test_specifyTargetComponet_BlueprintSpecified(self):
-        b = HexBlock("SodiumBlock", height=10.0)
-        sodiumDims = {"Tinput": 25.0, "Thot": 25.0, "op": 17, "ip": 0.0, "mult": 1.0}
-        ductDims = {"Tinput": 25.0, "Thot": 25.0, "op": 16, "ip": 15.0, "mult": 1.0}
-        dummy = Hexagon("coolant", "Sodium", **sodiumDims)
-        dummyDuct = Hexagon("duct", "FakeMat", **sodiumDims)
-        b.add(dummy)
-        b.add(dummyDuct)
+    def test_specifyTargetComponent_NotFound(self):
+        """ensure RuntimeError gets raised when no target component is found"""
+        b = HexBlock("fuel", height=10.0)
+        b.add(self.coolant)
+        b.setType("fuel")
+        with self.assertRaises(RuntimeError) as cm:
+            self.expData.determineTargetComponent(b)
+            the_exception = cm.exception
+            self.assertEqual(the_exception.error_code, 3)
+        with self.assertRaises(RuntimeError) as cm:
+            self.expData.determineTargetComponent(b, Flags.FUEL)
+            the_exception = cm.exception
+            self.assertEqual(the_exception.error_code, 3)
+
+    def test_specifyTargetComponent_singleSolid(self):
+        """ensures that specifyTargetComponent is smart enough to set the only solid as the target component"""
+        b = HexBlock("plenum", height=10.0)
+        ductDims = {"Tinput": 25.0, "Thot": 25.0, "op": 17, "ip": 0.0, "mult": 1.0}
+        duct = Hexagon("duct", "FakeMat", **ductDims)
+        b.add(duct)
+        b.add(self.coolant)
         b.getVolumeFractions()
-        b.setType("DuctBlock")
+        b.setType("plenum")
+        self.expData.determineTargetComponent(b)
+        self.assertTrue(
+            self.expData.isTargetComponent(duct),
+            msg=f"determineTargetComponent failed to recognize intended component: {duct}",
+        )
 
-        # check for no target component found
+    def test_specifyTargetComponent_viaType(self):
+        """set target component via type and not flags
+
+        Notes
+        -----
+        For the case in which a block contains a component that does not have Flags defined.
+        So instead of matching on flags, we match on type.
+        """
+        b = HexBlock("detector", height=10.0)
```

Comment:
@keckler @jakehader Here is the unit test that leverages the type checking. 

This block, "detector", contains a "detector" component. Since "detector" is not a registered flag the flag-based approach won't work. So our options are to either 1) modify source code to add a new flag, or 2) add a new capability to infer target components based on something other than flags. I got feedback that option 2 was more desirable and that checking types could be an option. 

The logic for `c.p.type in b.p.type or b.p.type in c.p.type` is to try and accommodate the case where the strings that define the component and block types aren't exactly the same. In this test, the component type is `detector neutron` and the block type is `neutron`. In this case the component type isn't "in" the block type, but the block type is "in" the component type. If the types were reversed, the opposite would be true (i.e., the component type would be "in" the block type).

Hopefully that's not as clear as mud...

Reply:
Isn't there a third option? To specify the target component on the blueprints using `axial expansion target component`? I'm just kinda confused why this change is necessary in the first place, but maybe I'm missing something.

Reply:
Yeah, that is a third, and very valid, option.

This PR is in response to a series of requests to not have to use that blueprints attribute.

Reply:
After some offline discussions, this type check was rolled back in https://github.com/terrapower/armi/commit/d9d138b9b40264ba9e82639e810235d67f12175d.